### PR TITLE
Results of running new "use nameof for nullable attributes" analyzer/fixer on roslyn

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-ide ptal.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs [hunk 1]
```diff
@@ -1126,7 +1126,7 @@ private BoundLambda ReallyBindForErrorRecovery()
                 ?? rebind(ReallyInferReturnType(delegateType: null, ImmutableArray<TypeWithAnnotations>.Empty, ImmutableArray<RefKind>.Empty));
 
             // Rebind a lambda to push target conversions through the return/result expressions
-            [return: NotNullIfNotNull("lambda")] BoundLambda? rebind(BoundLambda? lambda)
+            [return: NotNullIfNotNull(nameof(lambda))] BoundLambda? rebind(BoundLambda? lambda)
```
Comment (sharwell) [inline src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs:1129]: ❓ What's the scope here if a local variable has the same name?

**Review by sharwell — APPROVED** [review]

File: src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs:1129]: it's only the parameters.

**Comment by CyrusNajmabadi** [comment]
@jcouv @333fred could i get another set of compiler eyes on this?

**Review by 333fred — APPROVED** [review]

**Pull request merged**